sources/packages/api/src/services/student-account-applications/student-account-applications.service.ts

```diff
@@ -156,16 +156,18 @@ export class StudentAccountApplicationsService extends RecordDataModelService<St
   /**
    * Declines the student account application.
    * @param id student account application id.
+   * @param auditUserId user that should be considered the one that is causing the changes.
    */
-  async declineStudentAccountApplication(id: number): Promise<void> {
-    // TODO: To be changed to a soft delete.
-    const deleteResult = await this.repo
-      .createQueryBuilder()
-      .delete()
-      .from(StudentAccountApplication)
-      .where({ id })
-      .execute();
-    if (deleteResult.affected === 0) {
+  async declineStudentAccountApplication(
+    id: number,
+    auditUserId: number,
+  ): Promise<void> {
+    // Soft-delete the account application and associate the modifier.
+    const updateResult = await this.repo.update(id, {
```

Comment:
Just a suggestion. Can we create a wrapper softDelete() at DataModelService level? so that we can call that internal soft delete method and the wrapper method can set the deletedAt to new date.

Reply:
@ann-aot @henryoforeh-aot @guru-aot let's review this in some upcoming efforts for auditing. There are some considerations about how to have it better centralized but we can check how to do it better if possible.